bin/functions

```diff
@@ -1,7 +1,9 @@
 #!/usr/bin/env bash
+set -e
+set -o pipefail
```

Comment:
The script should fail if something goes wrong.

Reply:
https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

Reply:
Yes, unfortunately this didn't work out for now.
I would try to address this in a separate PR.